services/mailer/mail.go

```diff
@@ -220,3 +222,8 @@ func SendIssueMentionMail(issue *models.Issue, doer *models.User, content string
 	}
 	SendAsync(composeIssueCommentMessage(issue, doer, content, comment, mailIssueMention, tos, "issue mention"))
 }
+
+// SendIssueAssignedMail composes and sends issue assigned email
+func SendIssueAssignedMail(issue *models.Issue, doer *models.User, content string, comment *models.Comment, tos []string) {
```

Comment:
Could we consider a "issue was unassigned" notification? It's a less common scenario, though, unless it's an operation error from the part of the doer.

Reply:
Do you really want that by default? There is no refined mail settings that can make that an option for the user. At least for me, if I am not assigned any longer I probably don't care :)

Reply:
Is the assign/unassign mail only sent to assignee? it Maybe also sent to others?

Reply:
Issue was unassigned notification: I think it is a good idea but my opinion is it should be an option (in app.ini or elsewhere). It would be good if we can keep it out from this PR as it is already rather large.

Notification to others: I see this similar as the mention mail; being assigned is mostly interesting for the person being assigned. On the other hand update of assignees is a change of the issue, so it would make some sense to send notifications to all participants as well. 
The problem is that each added assignee is handled separate now, so if you add three new assignees it would trigger three notifications/mails. It would require a log of refactoring again to change that, and as I don't see it is very useful I would like to keep sending notification only to the assigned person.